src/go/k8s/controllers/redpanda/suite_test.go

```diff
@@ -153,6 +153,7 @@ var _ = BeforeSuite(func(done Done) {
 		err = k8sManager.Start(ctrl.SetupSignalHandler())
 		Expect(err).ToNot(HaveOccurred())
 	}()
+	Expect(k8sManager.GetCache().WaitForCacheSync(context.Background())).To(BeTrue())
```

Comment:
Can you write some context in the commit why this wait for cache call at the end of `BeforeSuite`? What race condition will this prevent? What state was out of sync?

Reply:
Updated